bench-tx/src/main.rs

```diff
@@ -0,0 +1,151 @@
+use std::{
+    fs::{read_to_string, write, File},
+    io::Write,
+    path::Path,
+};
+
+use miden_lib::{
+    notes::create_p2id_note, transaction::ToTransactionKernelInputs, utils::Serializable,
+};
+use miden_objects::{
+    accounts::AccountId,
+    assembly::ProgramAst,
+    assets::{Asset, FungibleAsset},
+    crypto::{dsa::rpo_falcon512::SecretKey, rand::RpoRandomCoin},
+    notes::NoteType,
+    transaction::TransactionArgs,
+    Felt,
+};
+use miden_tx::{TransactionExecutor, TransactionHost};
+use vm_processor::{ExecutionOptions, RecAdviceProvider, Word};
+
+mod utils;
+use utils::{
+    get_account_with_default_account_code, write_cycles_to_json, MockDataStore, String, ToString,
+    Vec, ACCOUNT_ID_FUNGIBLE_FAUCET_ON_CHAIN, ACCOUNT_ID_REGULAR_ACCOUNT_UPDATABLE_CODE_OFF_CHAIN,
+    ACCOUNT_ID_SENDER, DEFAULT_AUTH_SCRIPT,
+};
+
+pub enum Benchmark {
+    Simple,
+    P2ID,
+}
+
+fn main() -> Result<(), String> {
+    let path = Path::new("bench-tx/bench-tx.json");
+    let mut file = File::create(path).map_err(|e| e.to_string())?;
+    file.write_all(b"{}").map_err(|e| e.to_string())?;
+
+    // run all available benchmarks
+    benchmark_default_tx(path)?;
+    benchmark_p2id(path)?;
```

Comment:
I would probably have these functions return `TransactionBenchmark` and then write them both to JSON at the same time. Would probably simplify `write_cycles_to_json()` a bit.